File: 459-repeated-substring-pattern/459-repeated-substring-pattern.py
class Solution:
    def repeatedSubstringPattern(self, s: str) -> bool:
        i = 0
        n = len(s)
        string = ""
        if n == 1:
            return False
        for ch in s:
            string += ch
            i += 1
            if math.ceil(n / i) == math.floor(n / i):
                temp = string * (n // i)
                if temp == s:
                    return True
            if i == n // 2:
                break
        return False

File: 459-repeated-substring-pattern/459-repeated-substring-pattern.java
class Solution {
    public boolean repeatedSubstringPattern(String s) {
        int n;
        String str;
        str=s+s;
        n=str.length();
        str=str.substring(1, n-1);
        if(str.indexOf(s)!=-1)
            return true;
        return false;
    }
}